DUnitX.InternalDataProvider.pas

```diff
@@ -0,0 +1,30 @@
+unit DUnitX.InternalDataProvider;
+
+interface
+uses
+  DUnitX.Types;
+
+Type
+  ITestDataProvider = interface
```

Comment:
I'm not sure the interface is needed since you have a base class.

Reply:
Well, i did this, to seperate the methods used by the Manager/Factory from the methods used to create the testcases.

So, for example, if the Manager/Factory need a new method, the interface leaves untouched. So only the Baseclass and the Manager/Factory needs to be changed. The creation of the testcases do not need an update :)